src/ecsignature.js

```diff
@@ -71,7 +71,7 @@ ECSignature.parseScriptSignature = function (buffer) {
   var hashType = buffer.readUInt8(buffer.length - 1)
   var hashTypeMod = hashType & ~0x80
 
-  assert(hashTypeMod > 0x00 && hashTypeMod < 0x04, 'Invalid hashType ' + hashType)
+  if (hashTypeMod <= 0x00 || hashTypeMod >= 0x04) throw new Error('Invalid hashType ' + hashType)
```

Comment:
Moving away from `assert` as the string warning messages always get built, which has lead us to have various performance issue in the past.
It was nice while it lasted.